docs/DeveloperGuide.md

```diff
@@ -1,12 +1,251 @@
+
+A list function lists out all passengers' details in table form which is facilitated by `ListPassengerCommand`. It extends an abstract `Command` with an override method called `execute`.
+The abstract `Command` extends a `Parser` which holds and validates the User input to determine the type of command.
+
+</div>  
+
+Furthermore, `ListPassengerCommand` implements the following operation:
+* `execute(OperationList passengers, String lineInput)` &#8594; An override method inherited from `command` 
+to execute the list operation that should list out all the passengers from the logbook.
+
+<div align="justify">  
+The class diagram as depicted below shows the behaviour of the passenger list operation explained in two parts.  
+This scenario assumes that the manager have input a valid `passenger list` command for easier readability.  
+</div>
+
+Part 1:  
+
+* The valid `passenger list` command is passed as a `String lineInput` into the function `parse` to valid
+  the type of command that needs to be executed.
+* Once validated, a new Command called `ListPassengerCommand` should be instantiated.  
+
+![class diagram](../docs/ug-diagrams/images/passengerListcmdClassDiagram_0.jpg)  
+
+Part 2:  
+
+* `ListPassengerCommand` would then call `execute(passengers, lineInput)`,
+  associating itself with PassengerList to obtain the existing `ArrayList<FlightInfo> passengers`.
+* From there, `listOperation` method would run within method `execute` and
+prints out all `passengers` details on a table format.
+
+![class diagram](../docs/ug-diagrams/images/passengerListcmdClassDiagram_1.jpg)
+
+
+**Sequence diagram**
+
+![sequence diagram](../docs/ug-diagrams/images/passengerListcmdSeqDiagram.jpg)  
+
+When `paser` verifies that the command is an entity `passenger` and `list` operation,
+`ListPassengerCommend` is instantiated.  
+1. `execute(passenger, lineInput)` will run within `ListPassengerCommend`
+   which in turn would call on `listOperation()` in `PassengerList` class.
+2. Within `listOperation()`, it would first check if arraylist `passengers` obtained 
+from `OperationList` is empty or not.
+3. If `passengers` is empty, prints empty table, else prints respective passenger details in table form.
+
+---
+
+### Add a flight feature
+
+The Add a flight function adds a flight with its corresponding details to the flight list.
+
+**Sequence Diagram**
+
+**_NOTE:_** Exceptions are omitted for readability.
+
+![sequence diagram](../docs/ug-diagrams/images/flightAddCmdSeqDiagram.jpg)
+
+When the `Parser` recognizes the `add flight` command, `AddFlightCommand` is instantiated.
+1. The `AddFlightCommand` then implements a new `addOperation(lineInput:String)` in `FlightList`
+2. `FlightList` then instantiates the `FlightInfo` Object using the attributes retrieved from the 
+user input in the previous step.
+3. `Ui` class level method `showFlightAddedMessage()` is used to inform the user they have added a flight.
+
+
+---
+
+### Delete a flight feature
```

Comment:
All the written features were quite well written . Concise and easy to understand.